python/tests/test_tables.py

```diff
@@ -1009,6 +1009,23 @@ def test_set_with_optional_properties(self, codec):
             else:
                 assert md == row.metadata
 
+    def test_copy_metadata_schema(self):
+        table = self.table_class()
+        assert table.metadata_schema == tskit.MetadataSchema(None)
+        table.metadata_schema = tskit.MetadataSchema({"codec": "json"})
+        copy = table.copy()
+        table.assert_equals(copy)
+        # Check this independently to check the schema cache was invalidated
+        assert table.metadata_schema == copy.metadata_schema
```

Comment:
shouldn't you also have, say,
```
copy.metadata_schema = tskit.MetadataSchema(None)
assert table.metadata_schema != copy.metadata_schema
```

Reply:
well I suppose this is getting tested in the metadata schema tests.

Reply:
Added